app-react/src/App.tsx

```diff
@@ -1,26 +1,109 @@
-import React from 'react';
-import logo from './logo.svg';
-import './App.css';
+import { useEffect, useState } from 'react';
+import { Button, Stack, TextField, Typography } from '@mui/material';
+import SendIcon from '@mui/icons-material/Send';
+
+import { createTransaction, getServerStatus, getTransactions } from './api/requests';
+import { Transaction } from '../../app-express/src/api/types';
+import { TransactionView } from './components/TransactionView';
+
+const App = () => {
+  const [loading, setLoading] = useState(false);
+  const [serverIsUp, setServerIsUp] = useState(false);
+  const [transactions, setTransactions] = useState<Transaction[]>([]);
+  const [accountId, setAccountId] = useState('');
+  const [amount, setAmount] = useState('');
+
+  const orderedTransactions = transactions?.sort(
+    (a, b) => new Date(b.created_at).getTime() - new Date(a.created_at).getTime()
+  );
+
+  useEffect(() => {
+    const fetchData = async () => {
+      setLoading(true);
+      const ok = await getServerStatus();
+      setServerIsUp(ok);
+
+      const transactions = await getTransactions();
+      setTransactions(transactions);
+      setLoading(false);
+    };
+
+    fetchData();
+  }, []);
+
+  const createTransactionHandler = async () => {
+    setLoading(true);
+    const res = await createTransaction({ accountId, amount });
+
+    if (res.ok) {
+      setAccountId('');
+      setAmount('');
+
+      const transactions = await getTransactions();
+      setTransactions(transactions);
+    }
+    setLoading(false);
+  };
+
+  if (!serverIsUp) {
+    return (
+      <Stack p={6} spacing={6}>
+        <Typography>Server is down</Typography>
+      </Stack>
+    );
+  }
+
+  if (loading) {
+    return (
+      <Stack p={6} spacing={6}>
+        <Typography>Loading</Typography>
+      </Stack>
+    );
+  }
 
-function App() {
   return (
```

Comment:
I kind of strayed off from the html requirements in terms of using a form, hope it's ok!
As seen from the end users perspective, it should be fine at least